### PR TITLE
Feature/mainnet prep

### PR DESCRIPTION
**Pull request by talekhinezh**
* Remove betanet fork configs
* Cleanup deprecated logic

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/service/ActionParserService.java [hunk 1]
```diff
@@ -119,16 +119,12 @@ public ActionParserService(RriParser rriParser) {
...
+					validator(element)
 				).map(TransactionAction::register);
 
 			case UNREGISTER_VALIDATOR:
 				return allOf(
-					validator(element),
-					optionalName(element),
-					optionalUrl(element)
+					validator(element)
 				).map(TransactionAction::unregister);
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/service/ActionParserService.java:129]: I'll update open rpc spec with these changes

**Review by siy — APPROVED** [review]
LGTM

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/StakedValidatorsReducer.java [hunk 2]
```diff
@@ -75,42 +77,34 @@ public StakedValidatorsReducer(
...
 	@Override
 	public BiFunction<StakedValidators, Particle, StakedValidators> inputReducer() {
 		return (prev, p) -> {
-			if (p instanceof ValidatorParticle) {
-				var v = (ValidatorParticle) p;
-				if (v.isRegisteredForNextEpoch()) {
-					return prev.remove(v);
-				}
-			} else if (p instanceof PreparedStake) { // TODO: Remove for mainnet
+			if (p instanceof PreparedStake) { // TODO: Remove for mainnet
```
Comment (iamyulong) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/StakedValidatorsReducer.java:107]: Minor: This TODO can be removed now?

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/StakedValidatorsReducer.java [hunk 3]
```diff
@@ -75,42 +77,34 @@ public StakedValidatorsReducer(
...
 		return (prev, p) -> {
-			if (p instanceof ValidatorParticle) {
-				var v = (ValidatorParticle) p;
-				if (v.isRegisteredForNextEpoch()) {
-					return prev.add(v);
-				}
-				return prev;
+			if (p instanceof PreparedRegisteredUpdate) {
+				var v = (PreparedRegisteredUpdate) p;
+				return v.isRegistered() ? prev.add(v.getValidatorKey()) : prev.remove(v.getValidatorKey());
```
Comment (iamyulong) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/StakedValidatorsReducer.java:82]: Question: do we allow re-registering?

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/checkpoint/GenesisProvider.java [hunk 4]
```diff
@@ -121,6 +122,7 @@ public VerifiedTxnsAndProof get() {
 			// Initial validator registration
 			for (var validatorKey : validatorKeys) {
 				genesisTxnConstruction.registerAsValidator(validatorKey.getPublicKey());
+				genesisTxnConstruction.action(new UpdateAllowDelegationFlag(validatorKey.getPublicKey(), true));
```
Comment (iamyulong) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/checkpoint/GenesisProvider.java:125]: I thought there was a transition procedure which allows delegation if `AllowDelegationFlag` is missing, no?

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/berkeley/BerkeleyLedgerEntryStore.java [hunk 5]
```diff
@@ -527,18 +526,8 @@ public RawSubstateBytes next() {
...
-			return openIndexedCursorInternal(typeBytes.iterator().next(), deserialization);
-		} else if (typeBytes.size() == 2) {
-			var iter = typeBytes.iterator();
-			var cursor = openIndexedCursorInternal(iter.next(), deserialization);
-			return CloseableCursor.concat(cursor, () -> openIndexedCursorInternal(iter.next(), deserialization));
-		} else {
-			throw new IllegalStateException("Cannot handle more than 2 types per class");
-		}
+		var typeByte = deserialization.classToByte(particleClass);
+		return openIndexedCursorInternal(typeByte, deserialization);
```
Comment (iamyulong) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/berkeley/BerkeleyLedgerEntryStore.java:530]: Awesome!

File: radixdlt-engine/src/main/java/com/radixdlt/atom/REFieldSerialization.java [hunk 6]
```diff
@@ -61,6 +61,18 @@ public static ECDSASignature deserializeSignature(ByteBuffer buf) {
...
 	}
 
+	public static void serializeBoolean(ByteBuffer buf, boolean bool) {
+		buf.put((byte) (bool ? 1 : 0));
+	}
+
+	public static boolean deserializeBoolean(ByteBuffer buf) throws DeserializeException {
+		var flag = buf.get();
+		if (!(flag == 0 || flag == 1)) {
+			throw new DeserializeException("Invalid flag");
```
Comment (iamyulong) [inline radixdlt-engine/src/main/java/com/radixdlt/atom/REFieldSerialization.java:71]: Nice fix!

**Review by iamyulong — COMMENTED** [review]
Nice to see so many clean-ups! I will update the transaction specs to reflect the latest change.

**Review by LukasGasior1 — APPROVED** [review]

**Pull request merged**